### PR TITLE
GC: SharePoint: Backup: Abstract / Serialize

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Changes address updates to `sharePoint.Collection.Populate()`. 
- SharePoint Collections support `Lists` and `Pages`. Drives are supported in OneDrive at this time. 
- List's serialize function is abstracted to support `Pages`. 

Collection needs to support List and Pages support. Additionally, needs to use a similar interface as in `exchange` to ensure retry and async collection population

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* related to #2071 <issue>

## Test Plan
Test can be completed locally. Per Issue #2086, the Beta library interferes with overall caching. 
`TestSharePointPageCollection_Populate()` inspects changes

- [x] :zap: Unit test

### CODE REVIEW

File: src/internal/connector/sharepoint/collection.go [hunk 1]
```diff
@@ -176,51 +190,140 @@ func (sc *Collection) populate(ctx context.Context) {
...
 		close(colProgress)
-		sc.finishPopulation(ctx, objects, success, totalBytes, errs)
+		sc.finishPopulation(ctx, metrics.attempts, metrics.success, metrics.totalBytes, errs)
 	}()
 
-	// Retrieve list data from M365
+	// Switch retrieval function based on category
+	switch sc.category {
+	case List:
+		// do the thing
```
Comment (ryanfkeepers) [inline src/internal/connector/sharepoint/collection.go:199]: unnecessary comment

File: src/internal/connector/sharepoint/collection_test.go [hunk 2]
```diff
@@ -65,9 +65,9 @@ func (suite *SharePointCollectionSuite) TestSharePointDataReader_Valid() {
 	assert.Equal(t, readData, m)
 }
 
-// TestSharePointListCollection tests basic functionality to create
+// TestListCollection tests basic functionality to create
 // SharePoint collection and to use the data stream channel.
-func (suite *SharePointCollectionSuite) TestSharePointListCollection() {
```
Comment (ryanfkeepers) [inline src/internal/connector/sharepoint/collection_test.go:70]: Could use some tests for page collection here before this gets merged.

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/connector/sharepoint/collection.go (hunk 1 above)
Comment (dadams39) [reply src/internal/connector/sharepoint/collection.go]: Resolved with [ccd6ef8](https://github.com/alcionai/corso/pull/2187/commits/ccd6ef81a7477a91a2a407392f966e1ad01c025d)

File: src/internal/connector/sharepoint/collection_test.go (hunk 2 above)
Comment (dadams39) [reply src/internal/connector/sharepoint/collection_test.go:70]: Resolved with [d4de96f](https://github.com/alcionai/corso/pull/2187/commits/d4de96f6f578442570b18c75d9070cc73761db5e)

**Pull request merged**